testenv/aws/utils.go

```diff
@@ -0,0 +1,258 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	cloudformationtypes "github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+
+	"github.com/openclarity/vmclarity/testenv/utils"
+)
+
+func (e *AWSEnv) prepareStack(ctx context.Context) error {
+	var err error
+
+	// If the private and public key files are not provided, then generate a temporary key pair
+	if e.sshKeyPair.PublicKeyFile == "" || e.sshKeyPair.PrivateKeyFile == "" {
+		e.sshKeyPair, err = utils.GenerateSSHKeyPair(e.workDir)
+		if err != nil {
+			return fmt.Errorf("failed to generate ssh key pair: %w", err)
+		}
+	}
+
+	//	Read the public key file.
+	key, err := os.ReadFile(e.sshKeyPair.PublicKeyFile)
+	if err != nil {
+		return fmt.Errorf("failed to read public key: %w", err)
+	}
+
+	// Create a new key pair
+	_, err = e.ec2Client.ImportKeyPair(ctx, &ec2.ImportKeyPairInput{
+		KeyName:           aws.String(AWSKeyName),
+		PublicKeyMaterial: key,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to import key pair: %w", err)
+	}
+
+	// Create a new S3 bucket
+	_, err = e.s3Client.CreateBucket(ctx, &s3.CreateBucketInput{
+		Bucket: &e.stackName,
+		CreateBucketConfiguration: &s3types.CreateBucketConfiguration{
+			LocationConstraint: s3types.BucketLocationConstraint(e.region),
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create bucket: %w", err)
+	}
+
+	// Read template file
+	f, err := os.Open("../installation/aws/VmClarity.cfn")
```

Comment:
Please use the `installation.AWSManifestBundle` for the CFN tempate here. The same way as the docker-compose and helm chart based deployments does.